### PR TITLE
[AzureMonitorExporter] cleanup nullable in Exporter classes and Transmitter

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards #34013

## Changes
- remove `#nullable disable` in Exporter classes and resolve issues
- remove `#nullable disable` in Transmitter class and resolve issues

### Notes
The biggest change here is that the `AzureMonitorResource` in the `Exporter` could be null.
This is provided to the `TelemetryItem` constructor which is eventually used to set Tags.
The Tags were already checking for null so I think this change is safe.
https://github.com/Azure/azure-sdk-for-net/blob/b6eb01b26613b30808eddb5a26402613e03b714e/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs#L99-L102

### CODE REVIEW

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM.

@vishweshbankwar Could you please take a look and confirm.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 1]
```diff
@@ -64,7 +62,7 @@ public AzureMonitorTransmitter(AzureMonitorExporterOptions options, TokenCredent
             {
                 try
                 {
-                    var storageDirectory = options.StorageDirectory ?? StorageHelper.GetDefaultStorageDirectory();
+                    var storageDirectory = options.StorageDirectory ?? StorageHelper.GetDefaultStorageDirectory() ?? throw new InvalidOperationException("Unable to determine offline storage directory.");
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:65]: behavior change?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:65]: it looks like that, but not really a change.

`StorageHelper.GetDefaultStorageDirectory` can fail and return null.
That then causes the `new FileBlobProvider` constructor to throw.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:65]: but we were not throwing before right?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:65]: This throw does not crash the process. Catch block logs into event source. If this is acting as a blocker for this PR better to remove the changes and track it on an issue.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:65]: My bad - I was not reading it properly.

**Review by vishweshbankwar — APPROVED** [review]

**Pull request merged**